Sources/CoreData/Insert/DataRequest.swift

```diff
@@ -159,4 +159,68 @@ public extension DataRequest {
 			handler(response, save)
 		}
 	}
+
+	/// Adds a handler to be called once the request has finished.
+	///
+	/// - parameter type:              The `Insertable` type that will be used in the serialization
+	/// - parameter db:                The database to work in
+	/// - parameter queue:             The queue on which the deserializer (and your completion handler) is dispatched.
+	/// - parameter jsonOptions:       JSON decoder options
+	/// - parameter jsonTransformer:   The response JSON transformer
+	///
+	/// - returns: the response
+	@available(iOS 13, *)
+	func responseInsert<T: Insertable>(
+		of type: T.Type = T.self,
+		db: DB = DB.shared,
+		jsonOptions: JSONSerialization.ReadingOptions = .allowFragments,
+		jsonTransformer: @escaping (Any) throws -> Any = { $0 },
+		contextObject: Any? = nil,
+		automaticallyCancelling: Bool = false
+	) async -> DataResponse<T, AFError> {
+		let context = db.newBackgroundContext()
+		let save: DB.SaveBlockWitCallback = { completion in
+			context.perform {
+				do {
+					try context.save()
+					completion(nil)
+				} catch {
+					completion(error)
+				}
+			}
+		}
```

Comment:
Not sure how correct this will be. A big pitfall with async/await is when working with CoreData, and mixing threads/contexts. Do we have official examples somewhere on how to best handle this?